Library/Homebrew/cmd/update-report.rb

```diff
@@ -126,132 +127,143 @@ def output_update_report
+          hub.add(reporter, auto_update: args.auto_update?)
+        end
       end
 
-      begin
-        reporter = Reporter.new(tap)
-      rescue Reporter::ReporterRevisionUnsetError => e
-        onoe "#{e.message}\n#{e.backtrace.join "\n"}" if Homebrew::EnvConfig.developer?
-        next
-      end
-      if reporter.updated?
-        updated_taps << tap.name
-        hub.add(reporter, auto_update: args.auto_update?)
+      unless updated_taps.empty?
+        auto_update_header args: args
+        puts "Updated #{updated_taps.count} #{"tap".pluralize(updated_taps.count)} (#{updated_taps.to_sentence})."
+        updated = true
       end
-    end
 
-    unless updated_taps.empty?
-      auto_update_header args: args
-      puts "Updated #{updated_taps.count} #{"tap".pluralize(updated_taps.count)} (#{updated_taps.to_sentence})."
-      updated = true
-    end
-
-    if updated
-      if hub.empty?
-        puts "No changes to formulae." unless args.quiet?
-      else
-        if ENV.fetch("HOMEBREW_UPDATE_REPORT_ONLY_INSTALLED", false)
-          opoo "HOMEBREW_UPDATE_REPORT_ONLY_INSTALLED is now the default behaviour, " \
-               "so you can unset it from your environment."
-        end
+      if updated
+        if hub.empty?
+          puts "No changes to formulae." unless args.quiet?
+        else
+          if ENV.fetch("HOMEBREW_UPDATE_REPORT_ONLY_INSTALLED", false)
+            opoo "HOMEBREW_UPDATE_REPORT_ONLY_INSTALLED is now the default behaviour, " \
+                 "so you can unset it from your environment."
+          end
 
-        hub.dump(updated_formula_report: !args.auto_update?) unless args.quiet?
-        hub.reporters.each(&:migrate_tap_migration)
-        hub.reporters.each { |r| r.migrate_formula_rename(force: args.force?, verbose: args.verbose?) }
+          hub.dump(updated_formula_report: !args.auto_update?) unless args.quiet?
+          hub.reporters.each(&:migrate_tap_migration)
+          hub.reporters.each { |r| r.migrate_formula_rename(force: args.force?, verbose: args.verbose?) }
 
-        CacheStoreDatabase.use(:descriptions) do |db|
-          DescriptionCacheStore.new(db)
-                               .update_from_report!(hub)
-        end
-        CacheStoreDatabase.use(:cask_descriptions) do |db|
-          CaskDescriptionCacheStore.new(db)
-                                   .update_from_report!(hub)
+          CacheStoreDatabase.use(:descriptions) do |db|
+            DescriptionCacheStore.new(db)
+                                 .update_from_report!(hub)
+          end
+          CacheStoreDatabase.use(:cask_descriptions) do |db|
+            CaskDescriptionCacheStore.new(db)
+                                     .update_from_report!(hub)
+          end
         end
+        puts if args.auto_update?
+      elsif !args.auto_update? && !ENV["HOMEBREW_UPDATE_FAILED"] && !ENV["HOMEBREW_MIGRATE_LINUXBREW_FORMULAE"]
+        puts "Already up-to-date." unless args.quiet?
       end
```

Comment:
The `unless Homebrew::EnvConfig.disable_load_formula?` should end here since we want everything else to run even if `HOMEBREW_DISABLE_LOAD_FORMULA` is set

Reply:
See https://github.com/Homebrew/brew/pull/13482#discussion_r910311607